File: eCinema/eCinema.Core/Entities/Employee.cs
using eCinema.Core.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Core.Entities
{
    public class Employee : BaseEntity
    {
        public int ID { get; set; }
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;

        [DataType(DataType.Date)]
        public DateTime BirthDate { get; set; }
        public Gender Gender { get; set; }
        public bool IsActive { get; set; }
        public byte[]? ProfilePhoto { get; set; }

        [ForeignKey(nameof(Cinema))]
        public int CinemaId { get; set; }
        public Cinema Cinema { get; set; } = null!;
    }
}

File: eCinema/UI/ecinema_admin/lib/models/employee.dart
import 'package:ecinema_admin/models/cinema.dart';
import 'package:json_annotation/json_annotation.dart';

part 'employee.g.dart';

@JsonSerializable()
class Employee {
  int? id;
  String? firstName;
  String? lastName;
  String? email;
  String? phoneNumber;
  String? birthDate;
  int? gender;
  bool? isActive;
  String? profilePhoto;
  int? cinemaId;
  Cinema? cinema;

  Employee(this.id, this.firstName, this.lastName, this.email, this.phoneNumber, this.birthDate, this.gender,
      this.isActive, this.profilePhoto, this.cinemaId, this.cinema);

  factory Employee.fromJson(Map<String, dynamic> json) => _$EmployeeFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeToJson(this);
}
